thrift/server.go

```diff
@@ -79,13 +80,46 @@ func (s *Server) Register(svr TChanServer, opts ...RegisterOption) {
 	for _, m := range svr.Methods() {
 		s.ch.Register(s, service+"::"+m)
 	}
+	if registrar, ok := svr.(InterceptorRunnerRegistrar); ok {
+		registrar.RegisterInterceptorRunner(s)
+	}
 }
 
 // RegisterHealthHandler uses the user-specified function f for the Health endpoint.
 func (s *Server) RegisterHealthHandler(f HealthFunc) {
 	s.metaHandler.setHandler(f)
 }
 
+// RegisterInterceptor adds an interceptor to the server that will be
+// executed before and after each request
+func (s *Server) RegisterInterceptor(interceptor Interceptor) {
+	s.interceptors = append(s.interceptors, interceptor)
+}
+
+// RunPre executes the pre function intercepters registered on the
+// server until one of them fails, and produces a
+// InterceptorPostRunner that will excute the Post methods of all
+// inteceptors whose Pre methods were executed.
+func (s *Server) RunPre(ctx Context, method string, args thrift.TStruct) (InterceptorPostRunner, error) {
```

Comment:
noopPostRunner would need to return any incoming error, right? but yeah that sounds good